### ISSUE TITLE
Add support for Logout

### ISSUE THREAD

**Issue by jmelberg-okta** [okta/okta-sdk-appauth-ios#7]
Pending the release of OKTA-94106 terminate the user's session with Okta.

**Comment by luksfarris** [comment]
Hi @jmelberg-okta, do you have any plans to add this feature?
I'd love to contribute with a pull request. I assume the effort here is fetching the logout URL from the openid config, and calling ` - (BOOL)presentAuthorizationWithURL:LOGOUT_URL session:nil` from AppAuth

**Comment by jmelberg-okta** [comment]
Hey @luksfarris! I'd love to be able to get this in before our `1.0` release in the next few months. If you'd like to help out, please shoot over a PR! I'd be happy to incorporate it.

**Comment by luksfarris** [comment]
Hi @jmelberg-okta , turns out it's harder than I expected 😅. I've created PR https://github.com/openid/AppAuth-iOS/pull/191 for AppAuth, once it's merged I'll be able contribute here as well!

**Comment by luksfarris** [comment]
Hi @jmelberg-okta , the pull request I've created for AppAuth is moving forward, but more work will be needed in the future. Meanwhile, I've created a pull request for you that does the logout on our side, here it is #30 . I don't expect you to merge it soon, but at least this is something for us to start working with. It works for my okta setup

**Comment by ericraymond** [comment]
Hi @luksfarris, 

I extracted logoutFlow from your commit and modified it to call it directly from my app (not part of the SDK).  

I don't see any change in the logout behavior, the session still seems to be cached and the next login completes without any user interaction.

Any ideas on what I'm missing?

Thanks!

**Comment by andrewgotow** [comment]
Any progress on this?

I'm working on an enterprise application where users check out and share company devices, and official support for a log-out flow would be wonderful!

**Comment by omgitstom** [comment]
Thanks for your patience folks.

We are waiting on AppAuth (the underlying library that we use) to add this functionality:
https://github.com/openid/AppAuth-iOS/pull/259

Once this is merged and released, we will be in the position to update the Okta library.

**Comment by nbarbettini** [comment]
Closed via #103. We will be publishing a new version of this library soon with these changes, and the README has already been updated. If you need help with logout, let us know!
